File: src/main/java/br/com/correios/enderecador/cep/dnec/DnecCEPFactory.java
package br.com.correios.enderecador.cep.dnec;

import br.com.correios.enderecador.cep.CEPFactory;
import br.com.correios.enderecador.cep.InterfaceEnderecoDao;

public class DnecCEPFactory extends CEPFactory {
    private static DnecCEPFactory instance;

    public static synchronized DnecCEPFactory getIntance() {
        if (instance == null)
            instance = new DnecCEPFactory();
        return instance;
    }

    public InterfaceEnderecoDao getEndereco() {
        return EnderecoDnecDao.getInstance();
    }
}

File: src/main/kotlin/br/com/correios/enderecador/cep/dnec/DnecCEPFactory.kt
package br.com.correios.enderecador.cep.dnec

import br.com.correios.enderecador.cep.CEPFactory
import br.com.correios.enderecador.cep.InterfaceEnderecoDao
import kotlin.jvm.Synchronized

class DnecCEPFactory : CEPFactory() {
    override val endereco: InterfaceEnderecoDao?
        get() = EnderecoDnecDao.instance!!

    companion object {
        private var instance: DnecCEPFactory? = null

        @JvmStatic
        @get:Synchronized
        val intance: DnecCEPFactory?
            get() {
                if (instance == null) {
                    instance = DnecCEPFactory()
                }
                return instance
            }
    }
}
